fe/fe-core/src/main/java/com/starrocks/planner/OlapScanNode.java

```diff
@@ -707,6 +715,26 @@ public long getSelectedIndexId() {
         return selectedIndexId;
     }
 
+    private Set<Long> getHotPartitionIds(RangePartitionInfo partitionInfo, boolean canUseMultiversion) {
+        KeysType keysType = olapTable.getKeysType();
+        ConnectContext connectContext = ConnectContext.get();
+        Set<Long> hotIds = Sets.newHashSet();
+        if (connectContext == null) {
+            return hotIds;
+        }
+        List<Map.Entry<Long, Range<PartitionKey>>> partitions = partitionInfo.getSortedRangeMap(false);
+        int numHotIds = 0;
+        int numPartitions = partitions.size();
+        if (keysType == KeysType.PRIMARY_KEYS) {
+            int pkHotNum = connectContext.getSessionVariable().getQueryCachePrimaryKeysHotPartitionNum();
+            numHotIds = Math.min(numPartitions, Math.max(0, pkHotNum));
+        } else if (keysType == KeysType.UNIQUE_KEYS || (keysType == KeysType.AGG_KEYS && !canUseMultiversion)) {
```

Comment:
Why AGG_KEYS && ! canUseMultiversion needs this? is it uncacheable?

Reply:
Can we just use canUseMultiversion without depending on KeysType?

Reply:
No, we can not, primary key and unique key can not support multiversion cache, to see the comment.